File: python/AssignCookies.py
class AssignCookies(object):
    def findContentChildren(self,g,s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        if len(s)==0:
            return 0
        g.sort()
        s.sort()
        n=0
        i=0
        j=0
        while i<len(g):
            while j<len(s) and s[j]<g[i]:
                j+=1
            if j<len(s):
                n+=1
                i+=1
                j+=1
            else:
                break
        return n

File: Cplus/AssignCookies.cpp
#include <algorithm>
#include <vector>
using namespace std;

class Solution
{
public:
	int findContentChildren(vector<int> &g, vector<int> &s)
	{
		sort(g.begin(), g.end());
		sort(s.begin(), s.end());
		int res = 0, M = g.size(), N = s.size();
		for (int i = M - 1, j = N - 1; i >= 0 && j >= 0; --i)
		{
			if (s[j] >= g[i])
			{
				++res;
				--j;
			}
		}
		return res;
	}
};